File: frontend/src/app/services/status.ts
import { unwrapResponse } from "@/lib/appResponse";
import { Status } from "@wails/go/wails/AppHandler";

export type AppStatus = {
	ready: boolean;
	version: string;
};

/** アプリ全体の疎通状態を Wails binding から取得する。 */
export async function getAppStatus(): Promise<AppStatus> {
	const response = await Status();
	return unwrapResponse(response);
}

File: internal/handler/wails/status.go
package wails

import (
	"context"
	"log/slog"
)

// アプリ状態返却
func (h *AppHandler) GetStatus() Response[StatusResponse] {
	h.logger.Info(context.Background(), "app status requested", slog.String("operation", "status"))

	return OK(StatusResponse{
		Name:    "DB-checker",
		Ready:   true,
		Version: "dev",
	})
}
